### ISSUE TITLE
Number of unique contracts

### ISSUE THREAD

**Issue by ilblackdragon** [near/near-explorer#568]
Currently we should number of accounts with contracts and show how many are active.

Given we highly frequently leverage the factory pattern, where the same contract is deployed on many accounts (like staking pools, lockup, multisigs, etc). 

This doesn't allow to understand how many unique contracts does NEAR have.
We can just count dedupped on contract hash to show number of unique contracts on-chain.

This is especially interesting to compare between testnet and mainnet, seeing how much development gets converted into mainnet.

**Comment by frol** [comment]
We can use the `DEPLOY_CONTRACT` action to get the contract binary hash:

```sql
SELECT * FROM action_receipt_actions WHERE action_kind = 'DEPLOY_CONTRACT' LIMIT 1;
```

![image](https://user-images.githubusercontent.com/304265/109519790-55e3cc80-7a8a-11eb-97f4-cb0314b340dc.png)

There are currently 4868 unique `DEPLOY_CONTRACT` actions on testnet, and 117 on mainnet (over the whole history):

```sql
SELECT COUNT(DISTINCT args->>'code_sha256') FROM action_receipt_actions WHERE action_kind = 'DEPLOY_CONTRACT';
```

If we want to have it in a chart over time, we need to use something like this (pseudocode):

```sql
SELECT
    day,
    (SELECT COUNT(DISTINCT args->>'code_sha256')
       FROM action_receipt_actions
       WHERE  receipts.included_in_block_timestamp <= g.day
    ) AS dist_contract_deploys
FROM  (SELECT generate_series('2021-01-01'::date, '2021-01-06'::date, '1d')::date) AS g(day))
```

Basically, we will need to compute the unique value for each date. Alternatively, we may consider to pull all the `DEPLOY_CONTRACT` actions and compute the computation on Explorer backend side keeping the uniqueness accounting rolling.

**Comment by frol** [comment]
We can pull all the `DEPLOY_CONTRACT` events and aggregate them (iterate) on the Explorer backend (and cache it):

```sql
SELECT * FROM action_receipt_actions WHERE action_kind = 'DEPLOY_CONTRACT' ORDER BY included_in_block_timestamp;
```

**Comment by icerove** [comment]
from my understanding, for this unique contracts, we should separate it into two parts:
1. display daily unique deployed contracts amount (even though the contract can be deployed previously, at that day, it is unique deployed)
2. A list of unique contracts that on-chain for near( this definitely should be on a separate page /contracts, we can display the list totally there for now and then ask for design)
@frol @ilblackdragon

**Comment by icerove** [comment]
need help with Corwin if we display for the list of unique contract. @frol @corwinharrell

**Comment by frol** [comment]
Let's not overwhelm Corwin with minor tasks.

> display daily unique deployed contracts amount (even though the contract can be deployed previously, at that day, ...)

I don't think there is much use for this metric. I can see a value in (1) total contract deployments per date [non-unique], (2) unique contracts by a certain date (cumulative chart).

> page /contracts

That is a completely different story, orthogonal to this feature request, and it is going to be part of the /accounts page when we implement filtering.

**Comment by icerove** [comment]
@ilblackdragon @frol 
I still think making a page for every contracts we have for now make sense instead of giving filter on the accounts page and them filter contracts since that filter will be the accounts that have contracts with it instread showing the uniuque contracts that we have.

**Comment by ailisp** [comment]
From contract-runtime team, I think this metrics is interesting, this would quantify status of contract deploys and how many of them are unique, which we could use as basis for some optimizations. /cc @olonho
